python/ray/util/dask/callbacks.py

```diff
@@ -0,0 +1,211 @@
+                Run after submitting a Ray task.
+
+                Args:
+                    task (tuple): A Dask task, where the first tuple item is
+                        the task function, and the remaining tuple items are
+                        the task arguments (either the actual argument values,
+                        or Dask keys into the deps dictionary whose
+                        corresponding values are the argument values).
+                    key (str): The Dask graph key for the given task.
+                    deps (dict): The dependencies of this task.
+                    object_ref (ray.ObjectRef): The object reference for the
+                        return value of the Ray task.
+
+            - def _ray_pretask(key, object_refs):
+                Run before executing a Dask task within a Ray task. This
+                executes after the task has been submitted, within a Ray
+                worker. The return value of this task will be passed to the
+                _ray_posttask callback, if provided.
+
+                Args:
+                    key (str): The Dask graph key for the Dask task.
+                    object_refs (List[ray.ObjectRef]): The object references
+                        for the arguments of the Ray task.
+
+                Returns:
+                    A value that will be passed to the corresponding
+                    _ray_posttask callback, if said callback is defined.
+
+            - def _ray_posttask(key, result, pre_state):
+                Run after executing a Dask task within a Ray task. This
+                executes within a Ray worker. This callback receives the return
+                value of the _ray_pretask callback, if provided.
+
+                Args:
+                    key (str): The Dask graph key for the Dask task.
+                    result (object): The task result value.
+                    pre_state (object): The return value of the corresponding
+                        _ray_pretask callback, if said callback is defined.
+
+            - def _ray_postsubmit_all(object_refs, dsk):
+                Run after all Ray tasks have been submitted.
+
+                Args:
+                    object_refs (List[ray.ObjectRef]): The object references
+                        for the output (leaf) Ray tasks of the task graph.
+                    dsk (dict): The Dask graph.
+
+            - def _ray_finish(result):
+                Run after all Ray tasks have finished executing and the final
+                result has been returned.
+
+                Args:
+                    result (object): The final result (output) of the Dask
+                        computation, before any repackaging is done by
+                        Dask collection-specific post-compute callbacks.
+        """
+        for cb in CBS:
+            cb_func = kwargs.pop(cb, None)
+            if cb_func is not None:
+                setattr(self, "_" + cb, cb_func)
```

Comment:
I sacrificed some readability here to make the `CBS` tuple the source-of-truth for what Ray-specific callbacks exist.